code/game/objects/structures/dropship_equipment.dm

```diff
@@ -295,64 +295,180 @@
+	playsound(loc, 'sound/weapons/guns/interact/v51_load.ogg', 40, 1)
+	balloon_alert(user, "[LAZYLEN(loaded_grenades)]/[grenade_capacity] grenades loaded")
+
+/obj/structure/dropship_equipment/shuttle/nade_launcher/attack_hand(mob/living/user)
 	. = ..()
-	if(.)
+	if(!LAZYLEN(loaded_grenades))
 		return
-	if(istype(I, /obj/item/explosive/grenade/flare) && stored_amount < max_amount)
-		stored_amount++
-		user.balloon_alert(user, "You insert a flare, remaining flares [stored_amount].")
-		qdel(I)
 
-/obj/structure/dropship_equipment/shuttle/flare_launcher/update_equipment()
+	var/obj/item/explosive/grenade/first_nade = loaded_grenades[1]
+	user.put_in_hands(first_nade)
+	loaded_grenades -= first_nade
+	playsound(loc, 'sound/weapons/flipblade.ogg', 25, 1, 5)
+	balloon_alert(user, "[LAZYLEN(loaded_grenades)]/[grenade_capacity] grenades loaded")
+
+
+/obj/structure/dropship_equipment/shuttle/nade_launcher/update_equipment()
 	. = ..()
 	if(ship_base)
 		setDir(ship_base.dir)
 	else
 		setDir(initial(dir))
 	update_icon()
 
-/obj/structure/dropship_equipment/shuttle/flare_launcher/update_icon_state()
+/obj/structure/dropship_equipment/shuttle/nade_launcher/update_icon_state()
 	. = ..()
 	if(ship_base)
-		icon_state = "flare_system_installed"
+		icon_state = "nade_system_installed"
 	else
-		icon_state = "flare_system"
+		icon_state = "nade_system"
+
 
-///gets target and deploy the flare launcher
-/obj/structure/dropship_equipment/shuttle/flare_launcher/proc/deploy_flare()
+
+//////////////////////////////////// tanglefoot emitter //////////////////////////////////////
+
+/obj/structure/dropship_equipment/shuttle/tangle_emitter
+	equip_category = DROPSHIP_WEAPON
+	name = "disposable tanglefoot emitter"
+	desc = "A system that can emit tanglefoot while landing the aircraft to support aggressive landing positions. Can only be used once every ten minutes. Fits on the weapon attach points of dropships. You need a powerloader to lift it."
+	icon_state = "tfoot_system"
+	point_cost = 150
+	dropship_equipment_flags = IS_INTERACTABLE
+	/// Whether the system is currently enabled to activate on landing or not
+	var/enabled = TRUE
+	/// What type of smoke to use
+	var/obj/item/explosive/grenade/smokebomb/drain/pellet/pellet_type
+	/// Cooldown for emitting smoke
+	var/cooldown_length = 10 MINUTES
+	/// Identifier for system feedback
+	var/id = 0
+	COOLDOWN_DECLARE(use_cooldown)
+
+
+/obj/structure/dropship_equipment/shuttle/tangle_emitter/Initialize(mapload)
+	. = ..()
+	id = UNIQUEID
+
+/obj/structure/dropship_equipment/shuttle/tangle_emitter/equipment_interact(mob/user)
+	if(!enabled)
+		enabled = TRUE
+		update_icon()
+		user.balloon_alert(user, "You activate [src] #[id]. It will now deploy smoke when landing.")
+		RegisterSignal(linked_shuttle, COMSIG_MINIDROPSHIP_BEGIN_LANDING, PROC_REF(drop_pellet_to_location))
+		return
+	enabled = FALSE
+	update_icon()
+	user.balloon_alert(user, "You disable [src] #[id]. It will no longer deploy smoke when landing.")
```

Comment:
balloons too long: remember short and concise is best even if its not 100% correct english